app/events/zync_event.rb

```diff
@@ -53,6 +53,10 @@ def dependencies
     end
   end
 
+  def create_dependencies
+    dependencies.map { |dependency| ZyncEvent.create(self, dependency) }
```

Comment:
```suggestion
    dependencies.map { |dependency| self.class.create(self, dependency) }
```
Not very important, but just a small suggestion 😄